app/services/stripe_model_callbacks/invoice/updated_service.rb

```diff
@@ -1,23 +1,39 @@
 class StripeModelCallbacks::Invoice::UpdatedService < StripeModelCallbacks::BaseEventService
-  def execute
-    invoice.assign_from_stripe(object)
+  TRACKED_ACTIVITIES = {
```

Comment:
All this stuff with tracked activities hasn't anything to do with the locking issue. Please only do one thing per PR :-) 

I am not sure this simplifies anything. Before there were 4 really simple lines of code. Now there is a frozen hash at the top and some meta-code that checks if an activity should be created totalling 9 lines of code which is more than twice as much?

Reply:
I did that because of code duplication. It is just clean up, not a breaking change ;)

Reply:
As you can probably see from my comment, I am not under the impression this actually cleans up the code :-)

Reply:
The rest of the stuff you have done is awesome though :-)

Reply:
👍 I reverted that code.